### ISSUE TITLE
Room.flush is misleading and mostly useless

### ISSUE THREAD

**Issue by chobobobo** [ScreepsOCS/screeps.behaviour-action-pattern#1268]
This function leads people to believe that each room._ member persists between ticks when in fact all objects are destroyed between ticks.

If we really want to cache some of this information, we have to develop a mechanism to store this data in global and track it's validity between server instances.  For now all of the deletes should be removed as they will mislead others just like they have misled me.
```
mod.flush = function(){
    let clean = room => {
        delete room._sourceEnergyAvailable;
        delete room._droppedResources;
        delete room._ticksToNextRegeneration;
        delete room._relativeEnergyAvailable;
        delete room._towerFreeCapacity;
        delete room._hostiles;
        delete room._hostileIds;
        delete room._situation;
        delete room._casualties;
        delete room._currentCostMatrix;
        delete room._isReceivingEnergy;
        delete room._reservedSpawnEnergy;
        delete room._creeps;
        delete room._immobileCreeps;
        delete room._allCreeps;
        delete room._privateerMaxWeight;
        delete room._claimerMaxWeight;
        delete room._combatCreeps;
        delete room._defenseLevel;
        delete room._hostileThreatLevel;
        delete room._collapsed;
        delete room._feedable;
        console.log(room.name, 'avoid', room._avoidSKMatrix);
        if (room._avoidSKMatrix && Game.time >= room._avoidSKMatrix.validTick) {
            console.log(room.name, 'removing avoidSKMatrix', Game.time, room._avoidSKMatrix.validTick);
            delete room._avoidSKMatrix;
        }
        if( global.isNewServer ) {
            delete room._my;
            delete room._constructionSites;
            delete room._myConstructionSites;
            delete room._maxPerJob;
            delete room._minerals;
            delete room._structures;
        } else {
            delete room.structures._repairable;
            delete room.structures._urgentRepairableSites;
            delete room.structures._fortifyableSites;
            delete room.structures._fuelables;
        }
        if (!room._powerBank) {
            delete room.memory.powerBank;
        }
        room.newInvader = [];
        room.goneInvader = [];
    };
    Memory.observerSchedule = [];
    _.forEach(Game.rooms, clean);

    // Temporary migration can be removed once traveler is merged into /dev
    if (!_.isUndefined(Memory.rooms.hostileRooms)) {
        for (roomName in Memory.rooms.hostileRooms) {
            if (_.isUndefined(Memory.rooms[roomName])) Memory.rooms[roomName] = {};
            Memory.rooms[roomName].hostile = Memory.rooms.hostileRooms[roomName];
        }
        delete Memory.rooms.hostileRooms;
    }
};```
